contracts/RewardsLiquidator.vy

```diff
@@ -8,53 +8,63 @@ interface ERC20Decimals:
     def decimals() -> uint256: view
 
 
-interface LidoStethOracle:
-    def stethPrice() -> uint256: view
-
-
 interface ChainlinkAggregatorV3Interface:
     def decimals() -> uint256: view
     # (roundId: uint80, answer: int256, startedAt: uint256, updatedAt: uint256, answeredInRound: uint80)
     def latestRoundData() -> (uint256, int256, uint256, uint256, uint256): view
 
 
-interface CurveStableSwap:
+interface CurvePool:
     def get_dy(i: int128, j: int128, dx: uint256) -> uint256: view
     def exchange(i: int128, j: int128, dx: uint256, min_dy: uint256) -> uint256: payable
+    def exchange_underlying(i: int128, j: int128, dx: uint256, min_dy: uint256) -> uint256: nonpayable
 
 
 event SoldStethToUST:
     steth_amount: uint256
     eth_amount: uint256
+    usdc_amount: uint256
     ust_amount: uint256
-    steth_anchor_price: uint256
-    ust_anchor_price: uint256
+    steth_eth_price: uint256
+    eth_usdc_price: uint256
+    usdc_ust_price: uint256
 
 
 event AdminChanged:
     new_admin: address
 
 
 event PriceDifferenceChanged:
-    max_steth_price_difference_percent: uint256
-    max_eth_price_difference_percent: uint256
+    max_steth_eth_price_difference_percent: uint256
+    max_eth_usdc_price_difference_percent: uint256
+    max_usdc_ust_price_difference_percent: uint256
+    max_steth_ust_price_difference_percent: uint256
 
 
 UST_TOKEN: constant(address) = 0xa47c8bf37f92aBed4A126BDA807A7b7498661acD
+UST_TOKEN_DECIMALS: constant(uint256) = 18
+USDC_TOKEN: constant(address) = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48
+USDC_TOKEN_DECIMALS: constant(uint256) = 6
 STETH_TOKEN: constant(address) = 0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84
+STETH_TOKEN_DECIMALS: constant(uint256) = 18
 WETH_TOKEN: constant(address) = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2
+WETH_TOKEN_DECIMALS: constant(uint256) = 18
+
 
-LIDO_STETH_ETH_ORACLE: constant(address) = 0x3A6Bd15abf19581e411621D669B6a2bbe741ffD6
+CHAINLINK_STETH_ETH_FEED: constant(address) = 0x86392dC19c0b719886221c78AB11eb8Cf5c52812
+CHAINLINK_STETH_USD_FEED: constant(address) = 0xCfE54B5cD566aB89272946F602D76Ea879CAb4a8 #for final check
 CHAINLINK_ETH_USD_FEED: constant(address) = 0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419
+CHAINLINK_USDC_USD_FEED: constant(address) = 0x8fFfFfd4AfB6115b954Bd326cbe7B4BA576818f6
 
 CURVE_STETH_POOL: constant(address) = 0xDC24316b9AE028F1497c275EB9192a3Ea0f67022
-SUSHISWAP_ROUTER_V2: constant(address) = 0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F
+CURVE_UST_POOL: constant(address) = 0x890f4e345B1dAED0367A877a1612f86A1f86985f
+UNISWAP_ROUTER_V3: constant(address) = 0xE592427A0AEce92De3Edee1F18E0157C05861564
+UNISWAP_USDC_POOL_3_FEE: constant(uint256) = 3000 # 0.3%
```

Comment:
This should be extracted to a configurable state variable since a pool's fee might change and Uniswap router might fail to select any pool

Reply:
that's a moot point. Due to the fact that beside addresses of the pool contracts, there is hardcoded logic related to that pools (e.g. Curve StETH pool is StableSwap, but Curve USDC pool is MetaPool, and for each we've preset coin indices). I am afraid that simply replacing addresses is unlikely to be fully compatible with the current code, and it might be easier to redeploy the entire contract.

But in any case, I did the functions for setting the pools and their settings